internal/capability/types.go

```diff
@@ -27,9 +27,50 @@ type options struct {
 	reportWriter      io.Writer
 }
 
+type auditorOptions struct {
+	// AuditPlan holds the tests that should be run during an audit
+	auditPlan []string
+
+	// CatalogSource may be built-in OLM or custom
+	catalogSource string
+	// CatalogSourceNamespace will be openshift-marketplace or custom
+	catalogSourceNamespace string
+
+	// Packages is a subset of packages to be tested from a catalogSource
+	packages []string
+
+	// WorkQueue holds capAudits in a buffered channel in order to execute them
+	workQueue chan capAudit
+
+	// AllInstallModes will test all install modes supported by an operator
+	allInstallModes bool
+
+	// extraCustomResources associates packages to a list of Custom Resources (in addition to ALMExamples)
+	// to be audited by the OperandInstall AuditPlan.
+	extraCustomResources string
+
+	// OpCapClient is the main OpenShift client interface
+	opCapClient operator.Client
+
+	// Fs is an afero filesystem used by the auditor
+	fs afero.Fs
+
+	// Timeout is the audit timeout
+	timeout time.Duration
+
+	//  ReportWriter is any io.Writer for the text reports
+	reportWriter io.Writer
+}
+
 type (
 	auditFn        func(context.Context) error
 	auditCleanupFn func(context.Context) error
+
+	// auditOption is the function type for passing an option to an audit
+	auditOption func(options *options) error
+
+	// auditorOption is the function type for passing an option to RunAudits
+	auditorOption func(options *auditorOptions) error
```

Comment:
I would suggest to keep the names as obvious as possible. So options would be better as auditOptions making clear what are those for like auditorOptions. And both auditOption and auditorOption I would suggest to append Fn like auditOptionFn and auditorOptionFn or anything else that can make clear that those are functions to make it easily readable anywhere in the code for people that are not used to functional programming.

Reply:
I agree about renaming options to auditOptions. However, I'm on the fence about adding Fn... In some cases, I can see the utility, but here, I'm pretty well leaning toward not, as this is a fairly well-used pattern. If you can find me a case or two of functions options being used and calling the type OptionFn, then I'll reconsider.

Reply:
All right. It may be easy for you but certainly not to everybody. And I don't think we necessarily have to follow the mainstream every time. Specially if we agree on something that work for us. But anyway, that's just a suggestion not exactly a change request so no problem.